src/tools/fuzzing/fuzzing.cpp

```diff
@@ -1627,6 +1627,39 @@ void TranslateToFuzzReader::fixAfterChanges(Function* func) {
         i--;
       }
     }
+
+    void visitRethrow(Rethrow* curr) {
+      if (!isValidRethrow(curr->target)) {
+        replace();
+      }
+    }
+
+    bool isValidRethrow(Name target) {
+      if (controlFlowStack.empty()) {
+        return false;
+      }
+      Index i = controlFlowStack.size() - 1;
+      while (1) {
+        auto* curr = controlFlowStack[i];
+        if (auto* tryy = curr->dynCast<Try>()) {
+          // The rethrow must target a try, and must be nested in a catch of
+          // that try (not the body). Look at the child above us to check, when
+          // we find the proper try.
+          if (tryy->name == target) {
+            if (i + 1 >= controlFlowStack.size()) {
+              return false;
+            }
```

Comment:
Can't we just start from `controlFlowStack.size() - 2` for `i`?

Reply:
`- 2` might overflow, unless we add another check for that. I think the code is simpler as it is?

Reply:
Would changing `if (controlFlowStack.empty())` to `if (controlFlowStack.size() < 2)` in the beginning of the function be simpler? (With some comments, saying, there should be at least two control flow structures because there should be a `try` and a `block`)

Reply:
Makes sense, I guess we can just check for `< 2` there. Done.

But I don't follow the part about a block? It must be 2+ since there is the rethrow + a try.

Reply:
I thought there should be a `try` and an inner block for `try` body so that's 2.. No? `rethrow` itself isn't added to the CFG stack IIRC. Or should there always be a function-level block when we enter a function?

Reply:
Ok, looks like I was confused here between the control flow stack and the expression stack... hence my edited comment. I think I need to rework this code to fix that issue actually. Sadly this can't really be tested (we can't emit deterministic content from the fuzzer, not unless we somehow pick the right random bytes), so I didn't notice the problem using a test...

Reply:
Should be fixed now, I switched to ExpressionStackWalker, which is necessary for us to track the full parent path from the rethrow to the try.

If we always had a Block for each try child, we wouldn't need this, but we do allow anything in those places.